operator/apis/platform/v1alpha1/central_types.go

```diff
@@ -148,37 +148,63 @@ func (c *CentralComponentSpec) GetAdminPasswordGenerationDisabled() bool {
 	return pointer.BoolPtrDerefOr(c.AdminPasswordGenerationDisabled, false)
 }
 
-// CentralDBEnabled returns a bool if CentralDBSpec is not nil
+// CentralDBEnabled returns true if central db should be created
 func (c *CentralComponentSpec) CentralDBEnabled() bool {
-	if c == nil {
+	if c == nil || c.DB == nil || c.DB.Enabled == nil {
 		return false
 	}
-	return c.DB != nil
+
+	return *c.DB.Enabled == CentralDBEnabledTrue
 }
 
 // CentralDBSpec defines settings for the "central db" component.
 type CentralDBSpec struct {
+	// Specify whether central-db is enabled, Default configures central to use rocksdb.
+	// If Enabled the operator will provision the Central DB or use an external DB if a connection string is provided.
+	//+kubebuilder:validation:Default=Default
+	//+kubebuilder:default=Default
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=1
+	Enabled *CentralDBEnabled `json:"enabled,omitempty"`
```

Comment:
Renaming suggestion, in order to avoid confusion between attribute name `enabled` and enum value `Enabled`:
```suggestion
	IsEnabled *CentralDBEnabled `json:"isEnabled,omitempty"`
```

Reply:
Done.